symplyphysics/laws/thermodynamics/probability_of_ideal_gas_macrostate.py

```diff
@@ -0,0 +1,94 @@
+## by classical mechanics. Let us divide the volume of the vessel into `m` small enough cells with volumes `V_1`
+## upto `V_m`. Let us for a moment also attach a number to each gas particle to be able to tell them apart.
+
+## The _microstate_ of the gas is the complete description of (a) the particle count of each cell of the vessel
+## and (b) the particles of which numbers appear in each cell. A particle's moving within the same cell does not
+## change the microstate of the gas, whereas a particle's moving from one cell to another constitutes the change
+## of the microstate.
+
+## The _macrostate_ of the gas only requires the description of the particle count in each cell of the vessel.
+## If we assume the particles to be identical, it doesn't make a difference, from a macroscopic standpoint, particles
+## of which number appear in which cells. Hence the probability of the macrostate is larger than that of the microstate
+## by the factor of the number of permutations of the particles among the cells, which is exactly the statistical weight
+## of the macrostate.
+
+# Law: P = G * Product(p_i**N_i, i)
+## P - probability of macrostate
+## G - statistical weight of the macrostate
+## p_i - probability of finding one particle in i-th cell
+## N_i - number of particles in i-th cell
+
+# Conditions
+## - There are no external fields acting on the system
+## - All particles are identical
+## - Gas is ideal
+## - Probabilities must sum up to 1
+
+macrostate_probability = Symbol("macrostate_probability", dimensionless)
+statistical_weight = Symbol("statistical_weight", dimensionless)
+one_particle_in_cell_probability = SymbolIndexed("one_particle_in_cell_probability", dimensionless)
+particle_count_in_cell = SymbolIndexed("particle_count_in_cell", dimensionless)
+
+law = Eq(
+    macrostate_probability,
+    statistical_weight
+    * ProductIndexed(
+        one_particle_in_cell_probability[global_index]**particle_count_in_cell[global_index],
+        global_index,
+    ),
+)
+
+
+@validate_input(
+    one_particle_in_cell_probabilities_=one_particle_in_cell_probability,
+    particle_count_in_cells_=particle_count_in_cell,
+)
+@validate_output(macrostate_probability)
+def calculate_macrostate_probability(
+    one_particle_in_cell_probabilities_: Sequence[Probability],
+    particle_count_in_cells_: Sequence[int],
+) -> Probability:
+    if len(one_particle_in_cell_probabilities_) != len(particle_count_in_cells_):
+        raise ValueError("The number of particle counts must equal the number of probabilities")
+
+    try:
+        assert_equal(sum(one_particle_in_cell_probabilities_), 1)
+    except AssertionError as e:
+        raise ValueError("The probabilities must sum up to 1") from e
+
+    statistical_weight_ = factorial(sum(particle_count_in_cells_))
+    for particle_count_ in particle_count_in_cells_:
+        statistical_weight_ = statistical_weight_ // factorial(particle_count_)
```

Comment:
You could simply pass statistical weight here, instead of calculating it. Let's keep it this way, but put a link to a law, that justifies this calculation.

Reply:
I thought about it, but in my opinion this places an additional burden on the user to pre-calculate it. Maybe we can use an [already written function](https://github.com/blackyblack/symplyphysics/blob/0ca92cbcedea44b3388e78f5207f2e321780fa9f/symplyphysics/laws/thermodynamics/maxwell_boltzmann_statistics/statistical_weight_of_macrostate.py)?

Reply:
Better not. We tend to use loose coupling between the laws. So we only import other law, when current law depends on it, ie when deriving one law from another. Here, to maintain loose coupling we either explain, what we are doing here. Or accept statistical weight as parameter and let user decide, how to calculate it. First option looks better.